orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/cluster/FindImageFromClusterTask.groovy

```diff
@@ -119,6 +119,29 @@ class FindImageFromClusterTask extends AbstractCloudProviderAwareTask implements
     Set<String> imageNames = []
     Map<Location, String> imageIds = [:]
 
+    // Supplement config with regions from subsequent deploy steps:
+    def deployRegions = stage.execution.stages.findAll {
+      it.type == "deploy"
+    }.collect {
+      Set<String> regions = it.context?.clusters?.inject([] as Set<String>) { Set<String> accum, Map cluster ->
+        if (cluster.cloudProvider == stage.context.cloudProviderType) {
+          accum.addAll(cluster.availabilityZones?.keySet() ?: [])
+        }
+        return accum
+      } ?: []
+      if (it.context?.cluster?.cloudProvider == stage.context.cloudProviderType) {
+        regions.addAll(it.context?.cluster?.availabilityZones?.keySet() ?: [])
+      }
+      return regions
+    }.flatten()
+
+    deployRegions.forEach {
+      if (!config.regions.contains(it)) {
+        config.regions.add(it)
+        log.warn("Inferred and added region ($it) from deploy stage to FindImageFromClusterTask")
```

Comment:
Handy to get the `executionId` into the log message ... otherwise it's just lost in the noise.

(possibly also not a `warn`?)

Reply:
makes sense will add and change